src/EnergyPlus/SolarCollectors.cc

```diff
@@ -238,21 +235,16 @@ namespace SolarCollectors {
                     ErrorsFound = true;
                 }
 
-                {
-                    auto const SELECT_CASE_var(state.dataIPShortCut->cAlphaArgs(3));
-                    if (SELECT_CASE_var == "INLET") {
-                        state.dataSolarCollectors->Parameters(ParametersNum).TestType = TestTypeEnum::INLET;
-                    } else if (SELECT_CASE_var == "AVERAGE") {
-                        state.dataSolarCollectors->Parameters(ParametersNum).TestType = TestTypeEnum::AVERAGE;
-                    } else if (SELECT_CASE_var == "OUTLET") {
-                        state.dataSolarCollectors->Parameters(ParametersNum).TestType = TestTypeEnum::OUTLET;
-                    } else {
-                        ShowSevereError(state,
-                                        CurrentModuleParamObject + " = " + state.dataIPShortCut->cAlphaArgs(1) + ":  " +
-                                            state.dataIPShortCut->cAlphaArgs(3) + " is  not supported for " +
-                                            state.dataIPShortCut->cAlphaFieldNames(3));
-                        ErrorsFound = true;
-                    }
+                static const std::map<std::string, TestTypeEnum> testMap = {
```

Comment:
Don't do this. You're spending more time creating the map than you are actually doing lookups. Either create a `std::array<std::string_view>` and use `getEnumerationValue` or even leave the `if/else` ladder. We use `std::map` way too much. There are places where it makes sense to use them, but we use them in many places where it doesn't. This isn't python where everything is dynamic and on the heap anyway.

Reply:
I get it, the map is appealing, but using getEnumerationValue is a better approach.  I've made that change locally.

Reply:
Look [here](https://github.com/NREL/EnergyPlus/wiki/Containers%3A-When-to-Use-Which) for an explanation of when using `std::map` makes sense. The TL;DR version is that it makes sense for dynamic collections (i.e., collections that can only be created at runtime) and that have more than say 10 or 20 elements. For static collections, the advantages of `constexpr std::array` are overwhelming in my opinion. And for small dynamic collections, `std::vector` is probably cheaper. `std::map` is a good data structure for, say, mapping surface names to surface names (there can be 100s or 1000s of surfaces and their names are dynamic) or mapping zone names to zone indices (fewer zones than surfaces but if it's common enough to have more than 20 then it makes sense to use a map).